### ISSUE TITLE
PRO Install Fails at v3.1.1 due to a dependancy on the ConfigData.Version variable (PS Error - You cannot call a method on a null-valued expression)

### ISSUE THREAD

**Issue by pfoppe** [Esri/arcgis-powershell-dsc#299]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.1.1

### Affected Resource(s)

* PRO (confirmed)
* ArcGIS Desktop (probable)
* LicenseManager (probable)

### Configuration Files

Here is a sample config that works (it fails if you remove ConfigData.Version) - 
```
{
    "AllNodes": [
        {
            "NodeName": "<machine_name>",
			"Role": [
				"Pro"
            ]
    }
  ],
  "ConfigData": {
    "ProVersion": "2.6",
	"Version": "2.6",
    "Pro": {
      "AuthorizationType": "NAMED_USER",
      "BlockAddIns": "0",
      "SoftwareClass": "Professional",
      "AllUsers": "1",
      "EnableEUEI": "False",
      "CheckForUpdatesAtStartup": "False",
      "Installer": {
        "Path": "<Path_To_Installer>",
        "InstallDir": "C:\\Program Files\\ArcGIS\\Pro",
        "PatchesDir": "<Path_To_Patches>"
      }
    }
  }
}
```

### Expected Behavior

ArcGIS PRO should have been installed.  

### Actual Behavior

**Problem Statement -**  
Trying to run a new ArcGIS PRO install with ArcGIS PowerShell module v3.1.1 fails with the following exception: 
```
PS > Invoke-ArcGISConfiguration '.\myconfigfile.json' InstallLicenseConfigure -DebugSwitch
Dot Sourcing the Configuration:- ArcGISInstall
PSDesiredStateConfiguration\Node : You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\ArcGIS\3.1.1\Configurations-OnPrem\ArcGISInstall.ps1:22 char:5
+     Node $AllNodes.NodeName {
+     ~~~~
    + CategoryInfo          : InvalidOperation: (:) [PSDesiredStateConfiguration\node], ParentContainsErrorRecordExcep
   tion
    + FullyQualifiedErrorId : InvokeMethodOnNull,PSDesiredStateConfiguration\node

Compilation errors occurred while processing configuration 'ArcGISInstall'. Please review the errors reported in error
stream and modify your configuration code appropriately.
At
C:\Windows\system32\WindowsPowerShell\v1.0\Modules\PSDesiredStateConfiguration\PSDesiredStateConfiguration.psm1:3917
char:5
+     throw $ErrorRecord
+     ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (ArcGISInstall:String) [], InvalidOperationException
    + FullyQualifiedErrorId : FailToProcessConfiguration
```

### Steps to Reproduce

Setup a config file without the ConfigData.Version variable and run Invoke-ArcGISConfiguration
Add the ConfigData.Version variable and the execution will succeed.  

### Important Factoids

**Known Fix -** 
I was able to add the ConfigData.Version variable and set it to 2.6 and the installation succeeded.  Although please fix this bug (either update the documentation and sample configs to use this variable, or do not parse it where you are and only parse it for the roles that it applies to).  

### References

**Investigation -**  
Using the PSDebug, I was able to identify that the **ConfigData.Version** variable is being processed for ALL Node Roles even though it is not applicable for ArcGIS PRO (and Desktop): 
```
DEBUG:   85+          >>>> $VersionArray = $ConfigurationData.ConfigData.Version.Split(".")
DEBUG: 1031+                      >>>> $pscmdlet.CommandRuntime.WriteError((Get-InnerMostErrorRecord $_))
```

This appears to have been merged on this commit - [412bdaf5bbacbb08e0f222960681659e172177c1](https://github.com/Esri/arcgis-powershell-dsc/commit/412bdaf5bbacbb08e0f222960681659e172177c1#diff-73e008e4949abf61f2ba6ded1bc4f9dd8af8b073614d933830d6c3d7ec77b6c6R85)

Given the [Variables Reference Page](https://github.com/Esri/arcgis-powershell-dsc/wiki/V3.-Variables-reference-page-for-JSON-configuration-files), The "Version" variable only applies to ArcGIS Enterprise, ArcGIS WebStyles, ArcGIS Notebook Server, ArcGIS Mission Server.  Furthermore, the [Sample Config](https://github.com/Esri/arcgis-powershell-dsc/blob/master/SampleConfigs/v3/DesktopPro/Pro-Named.json) does not use the ConfigData.Version variable.

**Comment by cameronkroeker** [comment]
Hi @pfoppe,

Thanks for providing detailed information on this issue. This was an oversight and will be addressed in a future release. In the meantime the workaround you suggested will need to be used.

Thanks,
Cameron K.

**Comment by BrendanEsri** [comment]
Same thing happens with License Manager installs too

**Comment by cameronkroeker** [comment]
@pfoppe 

This issue has been addressed in v3.2.0.

https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v3.2.0

Thanks,
Cameron K.
